File: Interpreted/ackermann-function/ack.r

options(expressions=50000);
ackermann <- function(m, n)  {
   if ( m == 0 ) {
    n+1
  } else if ( n == 0 ) {
    ackermann(m-1, 1)
  } else {
    ackermann(m-1, ackermann(m, n-1))
  }
}

 print(ackermann(3, 9));

File: Interpreted/ackermann-function/ack.clj
(defn ackermann [m n]
  (cond (zero? m) (inc n)
        (zero? n) (ackermann (dec m) 1)
        :else (ackermann (dec m) (ackermann m (dec n)))))

(println (ackermann (Integer/parseInt (first *command-line-args* )) (Integer/parseInt (second *command-line-args* ))))
